# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpChannelChild.h
# Commit: 2289e685031e
# Full Hash: 2289e685031e878fee593b86cea90025b43d1fe9
# Author: Junior Hsu <juhsu@mozilla.com>
# Date: 2020-06-30 10:00:42
# Regressor Bug: 1633935
# File Overlap Count: 1
# Description:
#   Bug 1633935 - P10 wait PHttpChannel::OnStartRequestSent for permission/cookie update from parent, r=mayhemer,necko-reviewers
#   
#   Depends on D77751
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D77752
# ==============================================================================

diff -r 673030046319 -r 2289e685031e netwerk/protocol/http/HttpChannelChild.h
--- a/netwerk/protocol/http/HttpChannelChild.h	Tue Jun 30 00:33:54 2020 +0000
+++ b/netwerk/protocol/http/HttpChannelChild.h	Tue Jun 30 00:34:06 2020 +0000
@@ -132,6 +132,7 @@
   nsresult CrossProcessRedirectFinished(nsresult aStatus);
 
  protected:
+  mozilla::ipc::IPCResult RecvOnStartRequestSent() override;
   mozilla::ipc::IPCResult RecvFailedAsyncOpen(const nsresult& status) override;
   mozilla::ipc::IPCResult RecvRedirect1Begin(
       const uint32_t& registrarId, const URIParams& newURI,
@@ -472,6 +473,13 @@
   // CompleteRedirectSetup/RecvDeleteSelf.
   uint8_t mSuspendForWaitCompleteRedirectSetup : 1;
 
+  // True if RecvOnStartRequestSent was received.
+  uint8_t mRecvOnStartRequestSentCalled : 1;
+
+  // True if this channel is suspened by waiting for permission and cookie. That
+  // is, RecvOnStartRequestSent is received.
+  uint8_t mSuspendedByWaitingForPermissionAndCookie : 1;
+
   void FinishInterceptedRedirect();
   void CleanupRedirectingChannel(nsresult rv);
 